### PR TITLE
ENH matplotlib-base migration

### LINKED ISSUE #705: ENH add migrator to suggest dep on matplotlib-base instead of matplotlib
Per the discussion here (https://github.com/conda-forge/conda-forge.github.io/issues/933), we might want to suggest via a migration that people use a dep on matplotlib-base instead of matplotlib. 

IDK how many a packages this might be, but it could be feasible to do this by hand if there are a small number of them.

### PR DESCRIPTION
**Pull request by beckermr**
This PR adds the migration for `matplotlib` to `matplotlib-base`. Closes #705

### CODE REVIEW

**Comment by mariusvniekerk** [comment]
Should we make this slightly more generic as a package replacer migrator. This can be useful for other similar ones like jpeg-turbo, ruamel.yaml etc

**Comment by beckermr** [comment]
We can for sure. I won’t have time until later to make the changes.

**Comment by beckermr** [comment]
Alright @mariusvniekerk. This is ready for another look!

File: conda_forge_tick/auto_tick.xsh [hunk 1]
```diff
@@ -411,6 +411,52 @@ def add_rebuild_blas(migrators, gx):
...
+        bh = get_requirements(meta_yaml)
+        pkgs = set([old_pkg])
+        old_pkg_c = len(pkgs.intersection(bh)) > 0
+
+        rq = _host_run_test_dependencies(meta_yaml)
+
+        for e in list(total_graph.in_edges(node)):
+            if e[0] not in rq:
+                total_graph.remove_edge(*e)
+        if not any([old_pkg_c]):
```
Comment (CJ-Wright) [inline conda_forge_tick/auto_tick.xsh]: Isn't `old_pkg_c` a bool?

Comment (beckermr) [reply conda_forge_tick/auto_tick.xsh]: yeah it is. hold over from copy-paste. will change.

File: conda_forge_tick/auto_tick.xsh [hunk 2]
```diff
@@ -411,6 +411,52 @@ def add_rebuild_blas(migrators, gx):
...
+
+    """
+    total_graph = copy.deepcopy(gx)
+
+    for node, node_attrs in gx.nodes.items():
+        attrs = node_attrs['payload']
+        meta_yaml = attrs.get("meta_yaml", {}) or {}
+        bh = get_requirements(meta_yaml)
+        pkgs = set([old_pkg])
+        old_pkg_c = len(pkgs.intersection(bh)) > 0
```
Comment (CJ-Wright) [inline conda_forge_tick/auto_tick.xsh]: couldn't this just be `old_pkg in pkgs`? The current syntax would be useful if this supports multi-package replacement, but the code doesn't seem to support that.

File: conda_forge_tick/auto_tick.xsh [hunk 3]
```diff
@@ -411,6 +411,52 @@ def add_rebuild_blas(migrators, gx):
...
+                total_graph.remove_edge(*e)
+        if not old_pkg_c:
+            pluck(total_graph, node)
+
+    # post plucking we can have several strange cases, lets remove all selfloops
+    total_graph.remove_edges_from(nx.selfloop_edges(total_graph))
+
+    top_level = set(node for node in total_graph if not list(
+        total_graph.predecessors(node)))
+    cycles = list(nx.simple_cycles(total_graph))
```
Comment (CJ-Wright) [inline conda_forge_tick/auto_tick.xsh]:
```suggestion
```
This doesn't seem used.

File: conda_forge_tick/auto_tick.xsh [hunk 4]
```diff
@@ -411,6 +411,52 @@ def add_rebuild_blas(migrators, gx):
...
+        for e in list(total_graph.in_edges(node)):
+            if e[0] not in rq:
+                total_graph.remove_edge(*e)
+        if not old_pkg_c:
+            pluck(total_graph, node)
+
+    # post plucking we can have several strange cases, lets remove all selfloops
+    total_graph.remove_edges_from(nx.selfloop_edges(total_graph))
+
+    top_level = set(node for node in total_graph if not list(
```
Comment (CJ-Wright) [inline conda_forge_tick/auto_tick.xsh]: This doesn't seem used

**Comment by beckermr** [comment]
@CJ-Wright this code is all copy paste and me piecing things together. Hence all of the weird extra bits. I’m happy to clean it up more but given the amount we expect to use it, it’s not clear that is worth the time.

**Comment by CJ-Wright** [comment]
That's fair, although it will produce more work for the refactor. I'll accept this PR as is, but I'm not in a great position to keep an eye on it at the moment.

**Comment by beckermr** [comment]
This PR is not urgent. I am happy to wait until the refactor is done in order to not create more work.

**Comment by CJ-Wright** [comment]
@mariusvniekerk what do you think?

**Comment by beckermr** [comment]
bump here @mariusvniekerk

**Comment by CJ-Wright** [comment]
I think we decided this can go in before the refactor but I just haven't had time to baby sit it. If someone else has time please feel free to merge.

**Comment by beckermr** [comment]
Sounds good. LMK how I can help!

**Comment by CJ-Wright** [comment]
If I merge this, can you make certain that the circle_ci worker doesn't fall over? It runs every 5 minutes past the hour. Also can you use https://github.com/pulls?utf8=%E2%9C%93&q=is%3Aopen+is%3Apr+author%3Aregro-cf-autotick-bot+archived%3Afalse+ to check that the bot is opening PRs correctly? If something becomes a problem, please let me know an I can take action (opening a PR to disable this feature or revert the merge would be good).

Note to self: write guide to post merging stuff.

**Comment by beckermr** [comment]
yeah for sure - let me make sure I have access to everything before I fully commit - sit tight for a second!

**Comment by beckermr** [comment]
OK. I can see everything. Happy to go for it!

**Comment by CJ-Wright** [comment]
Great! Thank you very much!

**Pull request merged**